File: SchoolManagementApi/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using SchoolManagementApi.DTOs.Auth;
using SchoolManagementApi.Services;

namespace SchoolManagementApi.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AuthController(IAuthService authService) : ControllerBase
{
    private readonly IAuthService _authService = authService;

    [HttpPost("login")]
    public async Task<IActionResult> Login([FromBody] LoginDto loginDto)
    {
        if (string.IsNullOrWhiteSpace(loginDto.Email))
        {
            return BadRequest(new { error = "El correo es requerido." });
        }

        var result = await _authService.LoginAsync(loginDto.Email, loginDto.Role);

        if (result.Error)
        {
            return StatusCode(result.StatusCode, result);
        }

        return Ok(result);
    }
}

File: SchoolManagementApiNode/src/auth/auth.controller.ts
import { Controller, Post, Body } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { AuthService } from './auth.service';
import { LoginDto } from './dto/login.dto';

@ApiTags('Auth')
@Controller('auth')
export class AuthController {
  constructor(private readonly service: AuthService) {}

  @Post('login')
  @ApiOperation({ summary: 'Iniciar sesión' })
  login(@Body() dto: LoginDto) {
    if (!dto.email?.trim()) {
      return { statusCode: 400, message: 'El correo es requerido.', isSuccess: false, value: null };
    }
    return this.service.login(dto);
  }
}
